backend/app/api/v1/endpoints/cpt/maps/hce.py

```diff
@@ -17,35 +21,104 @@
-        df["ciSystem"] = "Jenkins"
-        df["testName"] = df["product"] + ":" + df["test"]
-        df["product"] = df["group"]
-        df["jobStatus"] = df["result"].apply(
-            lambda x: "SUCCESS" if x == "PASS" else "FAILURE"
-        )
-        df["version"] = df["version"].apply(
-            lambda x: x if len(x.split(":")) == 1 else x.split(":")[1][:7]
-        )
-        df["uuid"] = df["result_id"]
-        df["buildUrl"] = df["link"]
-        df["startDate"] = df["date"]
-        df["endDate"] = df["date"]
-        df = dropColumns(df)
-        return {"data": df, "total": response["total"]}
-    return {"data": pd.DataFrame(), "total": 0}
+
+    if isinstance(response, pd.DataFrame) or not response:
+        return {"data": pd.DataFrame(), "total": 0}
+    df = response.get("data", pd.DataFrame())
+    if df.empty:
+        return {"data": df, "total": response.get("total", 0)}
+
+    df["releaseStream"] = "Nightly"
+    df["ciSystem"] = "Jenkins"
+    df["testName"] = df["product"] + ":" + df["test"]
+    df["product"] = df["group"]
+    df["jobStatus"] = df["result"].apply(
+        lambda x: "SUCCESS" if x == "PASS" else "FAILURE"
+    )
+    df["version"] = df["version"].apply(
+        lambda x: x if len(x.split(":")) == 1 else x.split(":")[1][:7]
+    )
+    df["uuid"] = df["result_id"]
+    df["buildUrl"] = df["link"]
+    df["startDate"] = df["date"]
+    df["endDate"] = df["date"]
+    df = dropColumns(df)
+    return {"data": df, "total": response.get("total", 0)}
 
 
 def dropColumns(df):
     df = df.drop(
         columns=["group", "test", "result", "result_id", "link", "date", "release"]
     )
     return df
+
+
+async def hceFilter(start_datetime: date, end_datetime: date, filter: str):
+    try:
+        updated_filter = await get_updated_filter(filter)
+
+        response = await getFilterData(
+            start_datetime, end_datetime, updated_filter, f"hce.elasticsearch"
+        )
+
+        if isinstance(response, pd.DataFrame) or not response:
+            return {"total": 0, "filterData": [], "summary": {}}
+
+        if not response.get("filterData") or response.get("total", 0) == 0:
+            return {"total": response.get("total", 0), "filterData": [], "summary": {}}
+
+        # Add predefined filters
+        filters_to_add = [
+            {"key": "ciSystem", "name": "CI System", "value": ["Jenkins"]},
+            {"key": "releaseStream", "name": "Release Stream", "value": ["Nightly"]},
+        ]
+        response["filterData"].extend(filters_to_add)
+
+        # Filter data based on allowed keys
+        filtered_data = [
+            item for item in response["filterData"] if item.get("key") in keys_to_keep
+        ]
+
+        return {
+            "total": response.get("total", 0),
+            "filterData": filtered_data,
+            "summary": response.get("summary", {}),
+        }
+    except Exception as e:
+        print(f"Error retrieving filter data: {e}")
```

Comment:
Here's another place -- we're writing a message without any useful context on what happened, and then silently returning `None`.

If this is really unexpected, I'd much prefer that we *don't catch the exception*, and let FastAPI generate an HTTP 500 with a log traceback. If it's expected and should trigger an alternate API behavior, we need to implement that behavior.